pkg/querier/queryrange/codec.go

```diff
@@ -354,7 +363,7 @@ func (Codec) DecodeResponse(ctx context.Context, r *http.Response, req queryrang
 				Headers: httpResponseHeadersToPromResponseHeaders(r.Header),
 			}, nil
 		default:
-			return nil, httpgrpc.Errorf(http.StatusBadRequest, "unsupported response type")
+			return nil, httpgrpc.Errorf(http.StatusInternalServerError, "unsupported response type")
```

Comment:
This is changing an API - is this API exposed to end-users at all, or is it just internal?

Reply:
This should be a 500, and it was actually handle before in the marshalling. Now unmarshalling empty type won't error. This case was basically never possible before. So I just moved the validation here which makes more sense.

A test is covering this.